app/service/rest.py

```diff
@@ -609,6 +683,72 @@ def get_monthly_notification_stats(service_id):
     return jsonify(data=data)
 
 
+@service_blueprint.route(
+    "/<uuid:service_id>/notifications/<uuid:user_id>/monthly", methods=["GET"]
+)
+def get_monthly_notification_stats_by_user(service_id, user_id):
+    # check service_id validity
+    dao_fetch_service_by_id(service_id)
+    # user = get_user_by_id(user_id=user_id)
+
+    try:
+        year = int(request.args.get("year", "NaN"))
+    except ValueError:
+        raise InvalidRequest("Year must be a number", status_code=400)
+
+    start_date, end_date = get_calendar_year(year)
+
+    data = statistics.create_empty_monthly_notification_status_stats_dict(year)
+
+    stats = fetch_notification_stats_for_service_by_month_by_user(
+        start_date, end_date, service_id, user_id
+    )
+
+    statistics.add_monthly_notification_status_stats(data, stats)
+
+    now = datetime.utcnow()
```

Comment:
I wonder if our code analyzers could fail if they find deprecated methods in the code?

Reply:
I left the original deprecated method in some of the functions as I didn't want to disrupt any other code. I can update the time method across all functions when I make the other updates. @terrazoon presents a good point, we should be running a deprecated function check as part of the test coverage as some of this code is quite old.

Reply:
Oooh, that's a good question!  I don't know if `vulture` or `bandit` can help us here, or some other mechanism, because these don't throw deprecation warnings.  If they did, I think we could use some out-of-the-box tooling with Python itself to help catch that kind of stuff and maybe create another task in our `checks` GitHub Actions to find them, and/or a `make` command to use.

Either way, there's a bit of a code analysis step here for taking care of such stuff, so I'll add a note for that in the remove dead code or tech debt epics so we can tackle it at some point. 🙂

Reply:
I have changed this in all files in the notifications-api repo.